esp-hal-common/src/uart.rs

```diff
@@ -1252,18 +1297,8 @@ where
             return Ok(count);
         }
 
-        loop {
```

Comment:
I think an easier way to write this function is to block on checking `T::get_rx_fifo_count() > 0` then you can just call `drain_fifo` as we know we have at least one byte in the FiFo

Reply:
Good point. will do.